src/Block/AdminSearchBlockService.php

```diff
@@ -42,19 +42,59 @@ class AdminSearchBlockService extends AbstractBlockService
     protected $searchHandler;
 
     /**
-     * NEXT_MAJOR: Remove `$templating` argument.
+     * NEXT_MAJOR: Remove `$deprecatedTemplatingOrNull` argument.
      *
-     * @param Environment|string $twigOrName
+     * @param Environment|string $twigOrDeprecatedName
      */
-    public function __construct($twigOrName, ?EngineInterface $templating, Pool $pool, SearchHandler $searchHandler)
-    {
-        parent::__construct($twigOrName, $templating);
+    public function __construct(
+        $twigOrDeprecatedName,
+        ?EngineInterface $deprecatedTemplatingOrNull,
+        Pool $pool,
+        SearchHandler $searchHandler
+    ) {
```

Comment:
Can you rewrite it to allow use it in new way with andd `trigger_error` to old way? To allow create this Block in 2 ways:
```php
$block_v3 = new AdminSearchBlockService($name, $templating, $pool, $searchHandler);
$block_v4 = new AdminSerachBlockService($twig, $pool, $searchHandler);
```

Creating $block_v3 should trigger_error about deprecations.

Reply:
That change is out of the scope of this PR, which is giving support for BlockBundle 4.0, the signature was decided here: https://github.com/sonata-project/SonataAdminBundle/pull/5665 and it can be improved further on another PR.

With this PR you can already do what you mention like this:

```
$block_v3 = new AdminSearchBlockService($name, $templating, $pool, $searchHandler);
$block_v4 = new AdminSerachBlockService($twig, null, $pool, $searchHandler);
```

Reply:
So lets remove `trigger_error` from this PR It will be trigger by parent (AbstractBlockService)

Reply:
Done